src/Microsoft.AspNet.Mvc.Core/Routing/InnerAttributeRoute.cs

```diff
@@ -102,42 +101,35 @@ public class InnerAttributeRoute : IRouter
         /// <inheritdoc />
         public async Task RouteAsync([NotNull] RouteContext context)
         {
-            using (_logger.BeginScope("AttributeRoute.RouteAsync"))
+            foreach (var route in _matchingRoutes)
             {
-                foreach (var route in _matchingRoutes)
-                {
-                    var oldRouteData = context.RouteData;
+                var oldRouteData = context.RouteData;
 
-                    var newRouteData = new RouteData(oldRouteData);
-                    newRouteData.Routers.Add(route);
+                var newRouteData = new RouteData(oldRouteData);
+                newRouteData.Routers.Add(route);
 
-                    try
-                    {
-                        context.RouteData = newRouteData;
-                        await route.RouteAsync(context);
-                    }
-                    finally
+                try
+                {
+                    context.RouteData = newRouteData;
+                    await route.RouteAsync(context);
+                }
+                finally
+                {
+                    if (!context.IsHandled)
                     {
-                        if (!context.IsHandled)
-                        {
-                            context.RouteData = oldRouteData;
-                        }
+                        context.RouteData = oldRouteData;
                     }
+                }
 
-                    if (context.IsHandled)
-                    {
-                        break;
-                    }
+                if (context.IsHandled)
+                {
+                    break;
                 }
             }
 
-            if (_logger.IsEnabled(LogLevel.Verbose))
+            if (!context.IsHandled)
             {
-                _logger.WriteValues(new AttributeRouteRouteAsyncValues()
-                {
-                    MatchingRoutes = _matchingRoutes,
-                    Handled = context.IsHandled
-                });
+                _logger.LogVerbose("Request did not match any attribute route.");
```

Comment:
Is this consistent with what's going on in other routing pieces? I'll look at your other PR